Lib/idlelib/squeezer.py

```diff
@@ -210,6 +214,14 @@ def reload(cls):
             type="int", default=50,
         )
 
+        # Loading the font info requires a Tk root. IDLE doesn't rely
+        # on Tkinter's "default root", so the instance will reload
+        # font info using its editor windows's Tk root.
+        if cls._instance_weakref is not None:
+            instance = cls._instance_weakref()
```

Comment:
That was a bug! I've fixed the bug and improved `test_reload` for better coverage.

Reply:
I am still puzzled about the need to add weakrefs and tests. You said something earlier about "since the shell window can be closed and opened again.".  But if a shell window is closed, the associated squeezer and its buttons should be deleted.  If not, I would rather fix the bug than add a workaround.  Creating a new PyShell instance will create an new Squeezer instance.  What have I missed?